src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -398,6 +398,52 @@ class %s extends {$options['class']}
         return $code;
     }
 
+    private function isPreloadable(string $class): ?bool
+    {
+        static $cache = [];
+        if (array_key_exists($class, $cache)) {
+            return $cache[$class];
+        }
+        $cache[$class] = true; // prevent recursion
+
+        if (!class_exists($class) && !interface_exists($class, false) && !trait_exists($class, false)) {
+            return $cache[$class] = false;
+        }
+        $reflection = new \ReflectionClass($class);
+        if (!$reflection->isUserDefined()) {
+            return $cache[$class] = null;
+        }
+
+        // Since PHP 8.1, typehinted properties does not have to be auto-loadable.
+        // Before PHP 8.0, user can not define typehinted properties
+        if (\PHP_VERSION_ID >= 80100 || \PHP_VERSION_ID <= 80000) {
```

Comment:
the second comparison should use `<` instead of `<=`

Reply:
typed properties are a PHP 7.4 feature

Reply:
good catch